app/javascript/components/top-navbar/navbar-header.jsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch, useSelector } from 'react-redux';
+import { toggleVerticalMenuCollapsed } from '../../miq-redux/menu-reducer';
+
+const NavbarHeader = ({
+  customBrand,
+  imagePath,
+}) => {
+  const dispatch = useDispatch();
+  const isVerticalMenuCollapsed = useSelector(store => store.menuReducer.isVerticalMenuCollapsed);
+
+  return (
+    <div className="navbar-header">
+      <button
+        className="navbar-toggle"
+        type="button"
+        onClick={() => {
+          dispatch(toggleVerticalMenuCollapsed());
+          const content = window.document.getElementsByClassName('container-pf-nav-pf-vertical-with-sub-menus')[0];
+          if (isVerticalMenuCollapsed) {
+            content.classList.remove('collapsed-nav');
```

Comment:
I think it would be nice to check if the content exists. I don't know if its possible for the content no to be present in the DOM but its a good practice to first check if the element really exists before trying to call any functions on it.